e2e-tests/playwright/support/server/init.ts

```diff
@@ -42,9 +42,8 @@ export async function initSetup({
         const {client: userClient} = await makeClient(user);
 
         if (withDefaultProfileImage) {
-            // Set user profile image
-            const fullPath = path.join(path.resolve(__dirname), '../', 'asset/mattermost-icon_128x128.png');
-            await userClient.uploadProfileImageX(user.id, fullPath);
+            const imageData = getFileDataFromAsset('mattermost-icon_128x128.png', 'image/png');
+            await userClient.uploadProfileImage(user.id, imageData);
```

Comment:
Do we have any test case with `userClient.uploadFile`? To prevent any future regressions like this.

Reply:
We currently don't have and this is the closest, at least for the issue with FormData. However, I agree we should test it properly. With that, I've created a separate "client" folder/pattern that validates the response schema, its contents and other potential failure scenarios. PTAL.